### ISSUE TITLE
Add event Action or event Func where appropriate

### ISSUE THREAD

**Issue by beardordie** [VeryHotShark/First-Person-Controller-VeryHotShark#6]
This is just a suggestion. In my local branch I've added several events that are initialized as empty delegates. By adding these, it is easy to subscribe/unsubscribe to relevant events invoked by the FirstPersonController. In the following example I added events that will fire OnZoomIn and OnZoomOut from the CameraController class. I've done similar for OnJump, OnLand, OnBeginCrouch, OnEndCrouch, etc.

CameraController.cs additions:
![cameraController-1](https://user-images.githubusercontent.com/33188004/60312724-3689b900-9911-11e9-95cc-a182ca15a302.gif)
![cameraController-2](https://user-images.githubusercontent.com/33188004/60312732-3c7f9a00-9911-11e9-8d2b-535d7e824500.gif)

Then any other MonoBehaviour can subscribe to the events in OnEnable, and unsubscribe in OnDisable like this:
![newMonoBehaviour](https://user-images.githubusercontent.com/33188004/60312789-836d8f80-9911-11e9-9ba5-0e55606f90b9.gif)

**Comment by VeryHotShark** [comment]
Yes it a very good practice and way to structure a code :) can you somehow do a pull request so i can see other func and actions ?
